### PR TITLE
net/dns/resolver: add Windows ExitDNS service support, using net package

### PR DESCRIPTION
**Pull request by bradfitz**
Updates #1713
Updates #835

### CODE REVIEW

**Comment by bradfitz** [comment]
Not super efficient (thread per DNS query yet) but it works. I verified a Windows exit node can be a DNS relay now. We can fix DNS thread cost later, perhaps just in Go itself, moving Aaron's APC work there.

File: net/dns/resolver/tsdns.go [hunk 1]
```diff
@@ -926,6 +1132,26 @@ func (r *Resolver) respond(query []byte) ([]byte, error) {
...
+		s := strings.TrimSuffix(a, suf4)
+		// Parse and reverse octets.
+		ip, err := netaddr.ParseIP(s)
+		if err != nil || !ip.Is4() {
+			return "", false
+		}
+		a4 := ip.As4()
+		return netaddr.IPv4(a4[3], a4[2], a4[1], a4[0]).String(), true
+	}
+	// TODO: ipv6
```
Comment (sailorfrag) [inline net/dns/resolver/tsdns.go:1148]: I don't think this would be super hard, should this be done first?

**Review by sailorfrag — APPROVED** [review]

File: net/dns/resolver/tsdns_test.go [hunk 2]
```diff
@@ -1093,3 +1096,68 @@ func TestForwardLinkSelection(t *testing.T) {
 type linkSelFunc func(ip netaddr.IP) string
 
 func (f linkSelFunc) PickLink(ip netaddr.IP) string { return f(ip) }
+
+var hitNetwork = flag.Bool("hit-network", false, "run tests that require actually hitting the internet")
+
+func TestHandleExitNodeDNSQueryWithNetPkg(t *testing.T) {
+	if !*hitNetwork {
```
[the remark this answers is not in the document]
Comment (bradfitz) [reply net/dns/resolver/tsdns_test.go:1103]: On every platform but Windows. That's on my list to fix in Go 1.19.

File: net/dns/resolver/tsdns.go (hunk 1 above)
Comment (bradfitz) [reply net/dns/resolver/tsdns.go:1148]: oh right. Forgot. Will do.

File: net/dns/resolver/tsdns.go [hunk 3]
```diff
@@ -926,6 +1132,26 @@ func (r *Resolver) respond(query []byte) ([]byte, error) {
 	return marshalResponse(resp)
 }
 
+// unARP maps from "4.4.8.8.in-addr.arpa." to "8.8.4.4", etc.
```
Comment (dblohm7) [inline net/dns/resolver/tsdns.go:1135]: nit: s/unARP/unARPA/

File: net/dns/resolver/tsdns_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (bradfitz) [reply net/dns/resolver/tsdns_test.go]: Done.

[the remark this answers is not in the document]
Comment (bradfitz) [reply net/dns/resolver/tsdns_test.go]: (went a little overboard, but it was useful and found bugs!)

**Pull request merged**